tests/integration/test_sqs.py

```diff
@@ -1637,21 +1638,338 @@ def test_call_fifo_queue_url(self, sqs_client, sqs_create_queue):
+        assert queue_url.split("/")[-1] in response.text
 
-# TODO: test visibility timeout (with various ways to set them: queue attributes, receive parameter, update call)
-# TODO: test message attributes and message system attributes
+    @pytest.mark.aws_validated
+    def test_get_queue_attributes_with_query_args(self, sqs_create_queue, sqs_http_client):
+        queue_url = sqs_create_queue()
+        response = sqs_http_client.get(
+            queue_url,
+            params={
+                "Action": "GetQueueAttributes",
+                "AttributeName.1": "QueueArn",
+            },
+        )
 
+        assert response.ok
+        assert "<GetQueueAttributesResponse" in response.text
+        assert "<Attribute><Name>QueueArn</Name><Value>arn:aws:sqs:" in response.text
+        assert "<Attribute><Name>VisibilityTimeout</Name>" not in response.text
+        assert queue_url.split("/")[-1] in response.text
 
-class TestSqsLambdaIntegration:
-    pass
-    # TODO: move tests here
+    @pytest.mark.aws_validated
+    def test_get_queue_attributes_of_fifo_queue(self, sqs_create_queue, sqs_http_client):
+        queue_name = f"queue-{short_uid()}.fifo"
+        queue_url = sqs_create_queue(QueueName=queue_name, Attributes={"FifoQueue": "true"})
+
+        assert ".fifo" in queue_url
+
+        response = sqs_http_client.get(
+            queue_url,
+            params={
+                "Action": "GetQueueAttributes",
+                "AttributeName.1": "All",
+            },
+        )
+
+        assert response.ok
+        assert "<Name>FifoQueue</Name><Value>true</Value>" in response.text
+        assert queue_name in response.text
+
+    @pytest.mark.aws_validated
+    def test_get_queue_attributes_with_invalid_arg_returns_error(
+        self, sqs_create_queue, sqs_http_client
+    ):
+        queue_url = sqs_create_queue()
+        response = sqs_http_client.get(
+            queue_url,
+            params={
+                "Action": "GetQueueAttributes",
+                "AttributeName.1": "Foobar",
+            },
+        )
+
+        assert not response.ok
+        assert "<Type>Sender</Type>" in response.text
+        assert "<Code>InvalidAttributeName</Code>" in response.text
+        assert "<Message>Unknown Attribute Foobar.</Message>" in response.text
+
+    @pytest.mark.aws_validated
+    def test_get_delete_queue(self, sqs_create_queue, sqs_client, sqs_http_client):
+        queue_url = sqs_create_queue()
+
+        response = sqs_http_client.get(
+            queue_url,
+            params={
+                "Action": "DeleteQueue",
+            },
+        )
+        assert response.ok
+        assert "<DeleteQueueResponse " in response.text
+
+        def queue_is_removed():
+            try:
+                result = sqs_client.get_queue_url(QueueName=queue_url.split("/")[-1])
+                return result.get("QueueUrl") != queue_url
+            except ClientError:
+                return True
```

Comment:
Is it safe here to return True on any ClientError, or does this have the potential to be green on arbitrary issues?

Reply:
that's true, we should probably check whether it's a non-existent queue exception, good catch!